frontend/src/app/components/common/dialog/dialog.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { Observable } from 'rxjs';
+import { map, take } from 'rxjs/operators';
+import { DialogComponent } from './dialog.component';
+
+@Injectable()
+export class DialogService {
+  dialogRef?: MatDialogRef<DialogComponent>;
+
+  constructor(private dialog: MatDialog) { }
+
+
+  public open(title: string, message: string, cancelText: string, confirmText: string): void {
+    this.dialogRef = this.dialog.open(DialogComponent, {
+      data: {
+        title: title,
+        message: message,
+        cancelText: cancelText,
+        confirmText: confirmText
+      }
+    });
+  }
+  public confirmed(): Observable<any> | undefined {
```

Comment:
mixing observables and promises can be problematic, and in general promises are better because they support async/await. i'd reccomend calling `toPromise` on the observable and returning it as a promise so that consumers can await.